templates/aws/registry_nodes/terraform/pools/main.tf

```diff
@@ -0,0 +1,89 @@
+terraform {
+  required_version = ">= 0.13.1"
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = "~> 3.0"
+    }
+  }
+}
+
+provider "random" {}
+provider "aws" {
+  access_key = var.aws_access_key
+  secret_key = var.aws_secret_key
+  region =  var.aws_region
+}
+
+resource "random_id" "cluster_id" {
+  byte_length       = 6
+}
+
+resource "aws_key_pair" "corral_key" {
+  key_name       = "corral-${var.corral_user_id}-${random_id.cluster_id.hex}"
+  public_key = var.corral_public_key
+}
+
+data "cloudinit_config" "docker_service_config" {
+  gzip          = true
+  base64_encode = true
+  part {
+    content_type = "text/cloud-config"
+    content = file("${path.module}/cloud-init/setup-docker-service.yaml")
+  }
+}
+
+resource "aws_instance" "registry" {
+  ami = var.aws_ami
+  instance_type     = var.instance_type
+  key_name = aws_key_pair.corral_key.key_name
+  vpc_security_group_ids = [var.aws_security_group]
+  subnet_id = var.aws_subnet
+  user_data = data.cloudinit_config.docker_service_config.rendered
+
+  ebs_block_device {
+    device_name           = "/dev/sda1"
+    volume_size           = "200"
+    volume_type           = "gp3"
+    encrypted             = true
+    delete_on_termination = true
+  }
+
+  provisioner "remote-exec" {
+    inline = var.airgap_setup ? [
+      "sudo su <<EOF",
+      "echo ${var.corral_public_key} ${self.key_name} > /root/.ssh/authorized_keys",
+      "echo \"${var.corral_private_key}\" > /root/.ssh/id_rsa",
+      "chmod 700 /root/.ssh/id_rsa",
+      "EOF",
+    ]: [
+      "sudo su <<EOF",
+      "echo ${var.corral_public_key} ${self.key_name} > /root/.ssh/authorized_keys",
+      "EOF",
+    ]
+  }
+  connection {
+      type        = "ssh"
+      host        = self.public_ip
+      user        = var.aws_ssh_user
+      private_key = var.corral_private_key
+      timeout     = "4m"
+   }
+
+  tags = {
+    Name  = "${var.corral_user_id}-${random_id.cluster_id.hex}-registry"
+  }
+}
+
+resource "aws_route53_record" "aws_route53" {
+  zone_id            = data.aws_route53_zone.selected.zone_id
+  name               = "${aws_instance.registry.tags.Name}"
+  type               = "A"
+  ttl                = "300"
+  records            = [var.airgap_setup ? aws_instance.registry.private_ip : aws_instance.registry.public_ip]
```

Comment:
nit; instead of the private IP, you could use the public DNS. Since it is in the same subnet, in aws the private nodes can reach the public DNS name. This would allow the registry to be used by public setups as well, though that may be outside the scope of these corral tests.

Reply:
So you're saying even on an air gap setup using the DNS would work? I can do that, I was just mimicking what the current air gap job does.

Reply:
Sure, @izaac please confirm. For reference, we have the HA job (that uses a private IP) and the docker job (which uses  the DNS name). I don't have a strong preference for this, but thought I'd mention it.

Reply:
I think it'll depend. The job that uses the private IP generates the self signed certificates for the specific node's name `<name>.<region>.compute.amazonaws.com`

The other use the internal record of the `*.qa.rancher.space` and use valid certs. I guess it will depend on the certificate installation type. I'm inclined to either if it works for the cert type used.

Reply:
if internal IP will work for both cert types, it might be best just to leave it as you have it so as to not add additional complexity.